tools/f18/f18.cc

```diff
@@ -211,20 +208,17 @@ std::string CompileFortran(
   }
   if (driver.debugResolveNames || driver.dumpSymbols ||
       driver.dumpUnparseWithSymbols || driver.debugExpressions) {
-    std::vector<std::string> directories{options.searchDirectories};
-    directories.insert(directories.begin(), "."s);
-    if (driver.moduleDirectory != "."s) {
-      directories.insert(directories.begin(), driver.moduleDirectory);
-    }
-    (void)Fortran::semantics::ValidateLabels(parseTree, parsing.cooked());
-    Fortran::semantics::ResolveNames(Fortran::semantics::Scope::globalScope,
-        parseTree, parsing.cooked(), directories);
-    Fortran::semantics::ModFileWriter writer;
-    writer.set_directory(driver.moduleDirectory);
-    writer.WriteAll();
-    writer.errors().Emit(std::cerr, parsing.cooked());
+    semantics.Perform(parseTree);
+    auto &messages{semantics.messages()};
```

Comment:
Will warnings from semantics be sorted together with any parsing messages?

Reply:
Messages from parsing are emitted before doing any semantics processing. I didn't change that.

Reply:
OK.  We'll have to fix that later.